pages/dashboard.js

```diff
@@ -45,39 +45,136 @@ export default function Dashboard(props) {
 
   const [gisBenefit, setGisBenefit] = useState()
   const [gisLoaded, setGisLoaded] = useState(false)
-  const [gisError, setGisError] = useState(false)
+  const [gisError, setGisError] = useState()
 
   const [sebBenefit, setSebBenefit] = useState()
   const [sebLoaded, setSebLoaded] = useState(false)
   const [sebError, setSebError] = useState()
 
+  const [allBenefits, setAllBenefits] = useState()
+
   useEffect(() => {
-    async function fetchProgramData(program, setData, setError, setLoaded) {
-      try {
-        const res = await fetch(`/api/programData/${program}`)
-        if (res.ok) {
-          if (res.status === 200) {
-            setData(await res.json())
+    if (
+      !cppLoaded &&
+      !cppdLoaded &&
+      !eiLoaded &&
+      !oasLoaded &&
+      !gisLoaded &&
+      !sebLoaded
+    ) {
+      async function fetchProgramData(program, setData, setError, setLoaded) {
+        try {
+          const res = await fetch(`/api/programData/${program}`)
+          if (res.ok) {
+            if (res.status === 200) {
+              setData(await res.json())
+            }
+          } else {
+            const text = await res.text()
+            setError(`Error fetching ${program} data ${res.status} - ${text}.`)
           }
-        } else {
-          const text = await res.text()
-          setError(`Error fetching ${program} data ${res.status} - ${text}.`)
+        } catch (error) {
+          setError(`Something went wrong fetching ${program} data.`)
+          console.log(error)
+        } finally {
+          setLoaded(true)
         }
-      } catch (error) {
-        setError(`Something went wrong fetching ${program} data.`)
-        console.log(error)
-      } finally {
-        setLoaded(true)
       }
+
+      fetchProgramData('cpp', setCppBenefit, setCppError, setCppLoaded)
+      fetchProgramData('cppd', setCppdBenefit, setCppdError, setCppdLoaded)
+      fetchProgramData('ei', setEiBenefit, setEiError, setEiLoaded)
+      fetchProgramData('oas', setOasBenefit, setOasError, setOasLoaded)
+      fetchProgramData('gis', setGisBenefit, setGisError, setGisLoaded)
+      fetchProgramData('seb', setSebBenefit, setSebError, setSebLoaded)
     }
+  }, [cppLoaded, cppdLoaded, eiLoaded, gisLoaded, oasLoaded, sebLoaded])
+
+  useEffect(() => {
+    if (
+      cppLoaded &&
+      cppdLoaded &&
+      eiLoaded &&
+      oasLoaded &&
+      gisLoaded &&
+      sebLoaded
```

Comment:
I'm guessing this will block results showing on the page until all programs have returned data, is that right?
_It's probably okay to leave for now, but just want to be aware of it._

Reply:
Yes exactly. It gets really tricky handling it seperately and making all benefits show as soon as they are loaded individually. It would also make the page jump a lot if we display them as soon as some are loaded.

The current way its implemented will load them all together. Makes the page 'jump' only once.

Reply:
Yeah, we'll need a business decision on this at some point. If we do wait for all the data before showing this could be something we move to our API at some point.